### ISSUE TITLE
Apply current user settings on top of loaded command from Quick Launch dialog history

### ISSUE THREAD

**Issue by MaxRis** [jumptrading/FluentTerminal#230]
Global setting to use ConPty might be used to set actual state for Quick Launch commands loaded from the history.

**Comment by MaxRis** [comment]
There is also https://github.com/jumptrading/FluentTerminal/issues/229 to improve users` experience with commands history

**Comment by peske** [comment]
I think that it depends on requirements - what kind of behavior users want to see there. Let me try to explain the question:

Besides entering a command text (i.e. `mosh user@host`), user optionally can also change some options in the dialog: theme, tab color, EOL policy, ConPTY yes/no (which are obviously prefilled with defaults).

The question is what should happen when the user selects the same command from the history afterwards: should FT also use options used in the previous call (current behavior)? Or it should simply use current default values for these options?

Or to further clarify: when a command is executed for the first time, _Use ConPTY_ option is set to its default value, and user can change it if he wants. But when the same command is loaded from the history, _Use ConPTY_ will have the same value as it had on the last usage. It also stands for all other options (theme, tab color, etc.). I've created it this way because it makes sense in my opinion. But it can be changed...

Based on the answer we can change...

**Comment by mjs** [comment]
@peske I've thought about it a bit more and I think it makes sense to preserve the settings that went along with the saved command (i.e. current behaviour). Sometimes those options - especially ConPTY - can make a significant impact to terminal behaviour so it's important to preserve.

That's my 2c but it would be good to hear from actually users.

**Comment by MaxRis** [comment]
> Shouldn't they update each time the command is run?

Sounds reasonable.
Update should happen only if the input cmd line is the same, I believe, or how to track that the whole cmd input command is still the same after user's editing?
Also original Chris's request seems was to have fast way to switch between ConPty/WinPty for all command history at once.

**Comment by peske** [comment]
Regarding the bug noticed by @mjs (options being stuck to values used in the first use) - I've created a separate issue for that: https://github.com/jumptrading/FluentTerminal/issues/233

Regarding @cjw296' request - change in option globally should affect all commands, even if they are previously executed with some other option values (i.e. ConPTY yes/no), it also makes sense. For this reason I can suggest another variant (the third one):

Every config option in _Quick Launch_ dialog (and other dialogs for that matter) should have _use system default_ value. For example, "Use ConPTY" option would have three values:
- **Yes** - use ConPTY always, no matter currently active system settings;
- **No** - never use ConPTY, no matter currently active system settings;
- **System Default** - ConPTY will depend on the currently active system settings.

This would obviously satisfy @cjw296 request, but it would also make our UI bit more complicated.

My idea on how to implement this in UI is to add additional chekbox to every configuration option (theme, tab color, EOL handling, ConPTY yes/no), with meaning _system default_ (or maybe opposite _override default_). Depending on the selected value in this checkbox, the other control where actual value should be selected/entered, will be enabled/disabled.

Again, it solves the functional issue, but makes UI more cumbersome.

What do you think?

**Comment by MaxRis** [comment]
> Yes - use ConPTY always, no matter currently active system settings;
    No - never use ConPTY, no matter currently active system settings;
    System Default - ConPTY will depend on the currently active system settings.

Sounds pretty similar to what @mjs suggested during initial discussion. He also suggested idea to have dropdown or tristate control.

**Comment by peske** [comment]
Obviously I've missed your discussion... Anyway, it's here now for the record.

**Comment by mjs** [comment]
Yep this is what I proposed earlier. After some further thought I'm not convinced it's that great an idea as it makes the UI less clear. I'm currently leaning towards the current behaviour where options are stored with the command and the user has the opportunity to change things when reusing a historical command.

It would be good to hear from actual users though.

**Comment by cjw296** [comment]
So, the source of this was a real problem that's ended up with the current solution being quite cumbersome: ConPTY=on seems to universally cause problems, so a global default was added, however that only had any effect on hosts I'd never ssh'ed to before.

For every other host, I had to both remember to and then manually go and change the settings in the quick launch UI for every one of those hosts to ConPTY off. That was not fun.

So, what I think makes most sense for quick launch is "please only record options I've explicitly specified and use defaults for everything else, even when I come back to re-use a quick connect thing".

In my case, I only ever type the text command, usually `ssh user@host`. I don't think I've ever gone into the settings below that up until the ContPTY issue.

As a UI idea, don't add any extra checkboxes. You're right, that would make things more cumbersome. Just use the existing controls, but have nothing selected in them (indicating "use system default"), and have them take effect only when something is explicitly selected.

This isn't a blocking issue, but it can be quite annoying in circumstances where a global thing like ConPTY changes its default value.
